Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_wfsi.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <mbedtls/aes.h>
+
+#include "Common/CommonTypes.h"
+#include "Core/IPC_HLE/ESFormats.h"
+#include "Core/IPC_HLE/WII_IPC_HLE.h"
+#include "Core/IPC_HLE/WII_IPC_HLE_Device.h"
+
+class ARCUnpacker
+{
+public:
+  ARCUnpacker() { Reset(); }
+  void Reset();
+
+  void AddBytes(const std::vector<u8>& bytes);
+
+  using WriteCallback = std::function<void(const std::string&, const std::vector<u8>&)>;
+  void Extract(const WriteCallback& callback);
+
+private:
+  std::vector<u8> m_whole_file;
+};
+
+class CWII_IPC_HLE_Device_wfsi : public IWII_IPC_HLE_Device
+{
+public:
+  CWII_IPC_HLE_Device_wfsi(u32 device_id, const std::string& device_name);
+
+  IPCCommandResult Open(u32 command_address, u32 mode) override;
+  IPCCommandResult IOCtl(u32 command_address) override;
+  IPCCommandResult IOCtlV(u32 command_address) override;
+
+private:
+  std::string m_device_name;
+
+  mbedtls_aes_context m_aes_ctx;
+  u8 m_aes_key[0x10];
```

Comment:
You can just zero these out with

```cpp
 u8 m_aes_key[0x10] = {};
```

which will let you get rid of the memsets in the constructor.